src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const stateClone = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        Object.assign(stateClone, action.extraData);
+        break;
+      }
+
+      case 'removeProperties': {
+        for (const key of action.keysToRemove) {
+          delete stateClone[key];
+        }
+        break;
+      }
+
+      case 'clear': {
+        for (const key in stateClone) {
```

Comment:
Instead of using a loop to delete all properties, you can simply set 'stateClone' to an empty object. This will improve performance.